tensorbay/client/version.py

```diff
@@ -605,3 +607,103 @@ def delete_tag(self, name: str) -> None:
         delete_data: Dict[str, Any] = {"name": name}
 
         self._client.open_api_do("DELETE", "tags", self._dataset_id, json=delete_data)
+
+
+class JobMixin:  # pylint: disable=too-many-public-methods
+    """A mixin class supporting asynchronous jobs.
+
+    Arguments:
+        _dataset_id: Dataset ID.
+        _client: The client to interact between local and TensorBay.
+        _status: The version control status of the dataset.
+
+    """
+
+    _dataset_id: str
+    _client: Client
+    _status: Status
+
+    def _create_job(  # pylint: disable=too-many-arguments
+        self,
+        title: str,
+        job_type: str,
+        arguments: Dict[str, Any],
+        description: str = "",
+        limit_retry: int = 3,
+    ) -> str:
+        """Create a :class:`Job`.
+
+        Arguments:
+            title: The SquashAndMergeJob title.
+            job_type: The type of Job.
+            arguments: The arguments dict of the specific job.
+            description: The SquashAndMergeJob description.
+            limit_retry: The limit retry times of the SquashAndMergeJob.
+
+        Returns:# flake8: noqa: DAR202
+            The id of the job.
+
+        """
+
+    def _get_job(self, job_id: str) -> Dict[str, Any]:
+        """Get a :class:`Job`.
+
+        Arguments:
+            job_id: The Job id.
+
+        Returns:# flake8: noqa: DAR202
+            The info of Job.
+
+        """
+
+    def create_squash_and_merge_job(
+        self,
+        title: str,
+        description: str = "",
+        limit_retry: int = 3,
+        *,
+        draft_title: str,
+        source_branch_name: str,
+        draft_description: str = "",
+        target_branch_name: Optional[str] = None,
+        strategy: Optional[str] = "abort",
```

Comment:
should the argument order in function signature and in the docsting be the same?

Also, should we put the `source_branch_name` and `target_branch_name` together in the signature.